spec/requests/posts_spec.rb

```diff
@@ -0,0 +1,24 @@
+require 'rails_helper'
+
+RSpec.describe 'Post', type: :request do
+  describe 'Get index route of users controller' do
```

Comment:
Post Spec file should include tests for both `show` and `index` that ensures correct view template is rendered.
`Example : expect(response).to render_template(:show)`

Post Spec file should include test for both `show` and `index` that ensures the response body includes the correct placeholder text.
`Example : expect(response.body).to include("Welcome to user index page")`